include/caliper/Annotation.h

```diff
@@ -202,6 +207,39 @@ class Annotation
     /// \copydoc cali::Annotation::begin(int)
     Annotation& begin(const Variant& data);
 
+#ifdef CALI_FORWARDING_ENABLED
+    template<typename Arg, typename... Args>
+    struct head{
+        using type = Arg;
+    };
+
+    template<typename... Args>
+    auto begin(Args&&... args) -> typename std::enable_if<!std::is_same<typename head<Args...>::type,cali::Variant>::value,Annotation&>::type {
+        return begin(Variant(std::forward<Args>(args)...));
+    }
+    template<typename... Args>
+    auto set(Args&&... args) -> typename std::enable_if<!std::is_same<typename head<Args...>::type,cali::Variant>::value,Annotation&>::type {
+        return set(Variant(std::forward<Args>(args)...));
+    }
+#else
+    template<typename Arg>
+    Annotation& begin(const Arg& arg){
+        return begin(arg);
```

Comment:
Don't we need `begin(Variant(arg))` here?

Reply:
Yiiiikes

Reply:
(Will fix)